doc/source/notebooks/gpflow_2_upgrade/gpflow2_upgrade_guide.ipynb

```diff
@@ -0,0 +1,299 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use of tf.function\n",
+    "\n",
+    "Wrapping compute-heavy operations such as calculating a model objective or even the optimizer steps (such as `tf.optimizers.Adam().minimize()`) with `tf.function` is crucial for efficient computation. \n",
+    "\n",
+    "**Note**: you should ensure that functions wrapped in `tf.function` are only passed **tensors** (not numpy arrays or other data structures, with the exception of a small number of bool or enum-style flags), or the decorator will re-compile the graph each time the function is passed new objects as its arguments. See the [TensorFlow documentation on re-tracing](https://www.tensorflow.org/guide/function#re-tracing) for further details.\n",
+    "\n",
+    "You can convert a numpy array to a tensor by using `tf.constant`. For example: `compiled_function(tf.constant(numpy_array))`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model Compilation\n",
+    "\n",
+    "Models no longer need to be compiled before use. Remove all calls to the `compile` method. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Sessions and Graphs\n",
+    "\n",
+    "GPflow only supports eager execution, which is the default in TensorFlow 2. It does not support graph mode, which was the default execution mode in TensorFlow 1. Therefore all references to Sessions and Graphs should be removed. You should also remove references to the `gpflow.reset_default_graph_and_session` function.\n",
+    "\n",
+    "**Warning**: code that creates graphs (for example `tf.Graph().as_default()`) will disable eager execution, which will not work well with GPflow 2. If you get errors like “'Tensor' object has no attribute 'numpy'” then you may not have removed all references to graphs and sessions in your code."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Defer Build\n",
+    "\n",
+    "The `defer_build` context manager has been removed. References to it can simply be removed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Return Types from Auto-flowed Methods\n",
+    "\n",
+    "GPflow methods that used the `@autoflow` decorator, like for example `predict_f` and `predict_y`, will previously have returned NumPy Arrays. These now return TensorFlow tensors. In many cases these can be used like NumPy arrays (they can be passed directly to many of NumPy’s functions and even be plotted by matplotlib), but to actually turn them into numpy arrays you will need to call `.numpy()` on them.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"numpy_new.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Parameter Values\n",
+    "\n",
+    "GPflow’s `Parameter.value` has changed from a property to a method.\n",
```

Comment:
I think that it would great to add that actually `value` is not needed anymore because of the Parameter nature; it just behaves as a TensorFlow Tensor.

Reply:
> I think that it would great to add that actually `value` is not needed anymore because of the Parameter nature; it just behaves as a TensorFlow Tensor.

I have mentioned this